lesson_07/objects/objects_app/src/main/java/com/codedifferently/lesson7/richhawkins/House.java

```diff
@@ -0,0 +1,61 @@
+package com.codedifferently.lesson7.richhawkins;
+
+import java.time.Year;
+
+public class House {
+  //     The class must have at least 5 member variables of at least 3 different types.
+  String streetName;
+  int streetNum;
+  int yearBuilt;
+  boolean hasGarage;
+  short numberOfBedrooms;
+
+  // You must also incorporate an enum type as well.
+
+  public enum HouseType {
+    APARTMENT,
+    BUNGALOW,
+    COTTAGE,
+    CAPECOD,
+    COLONIAL,
+    FARMHOUSE,
+    MANSION,
+    TINYHOME,
+    TOWNHOUSE,
+    SINGLEFAMILY,
+    SPLITLEVEL,
+  }
+
+  // The class must have at least one constructor.
+
+  public House(String streetName, int streetNum) {
+    this.streetName = streetName;
+    this.streetNum = streetNum;
+  }
+
+  // The class must have at least 3 member functions.
+
+  // One of your functions must make use of a conditional expression.
+  // You must use at least one custom exception.
+  public boolean checkValidYear(int yearBuilt) {
```

Comment:
I'd expect this method to be `private` and to be used in a setter property called `setYearBuilt` that allows you to set a proper year or throw an exception if the year is not valid. Please revise.

Reply:
This really helped make a few things click for me. Should be revised as requested.